### PR TITLE
[OPIK-312] Add Get prompt by id

### PR DESCRIPTION
**Pull request by thiagohora**
## Details
Add Get prompt by id

## Issues
[OPIK-312

### CODE REVIEW

File: apps/opik-backend/src/main/java/com/comet/opik/utils/MustacheVariableExtractor.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    public static final MustacheFactory MF = new DefaultMustacheFactory();
+
+    public static Set<String> extractVariables(String template) {
+        Set<String> variables = new HashSet<>();
+
+        // Initialize Mustache Factory
+        Mustache mustache = MF.compile(new StringReader(template), "template");
+
+        // Get th e root node of the template
```
Comment (idoberko2) [inline apps/opik-backend/src/main/java/com/comet/opik/utils/MustacheVariableExtractor.java]: typo
```suggestion
        // Get the root node of the template
```

File: apps/opik-backend/src/main/java/com/comet/opik/utils/MustacheVariableExtractor.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        Code[] codes = mustache.getCodes();
+        collectVariables(codes, variables);
+
+        return variables;
+    }
+
+    private static void collectVariables(Code[] codes, Set<String> variables) {
```
Comment (idoberko2) [inline apps/opik-backend/src/main/java/com/comet/opik/utils/MustacheVariableExtractor.java]: I see that this function is recursive and probably that's why you chose to mutate the variables argument but I think it's better to have another helper function that encapsulates the recursiveness and change this function signature to be pure

Comment (thiagohora) [reply apps/opik-backend/src/main/java/com/comet/opik/utils/MustacheVariableExtractor.java]: I'm not sure about the benefit of it. If you see, this function is private, and the Set is created internally, so there is no leaking of external references' mutations.  Making the function pure here would be important if we use some sort of concurrency. But as it's now, we would only consume more memory. I mean, I would understand it if the Set was received as an argument of the public function, but in this case, it's created internally. I'm not sure I'm missing something; if you want, we can have a call tomorrow to discuss it better.

**Review by idoberko2 — APPROVED** [review]
The benefit would be easier to change in the future, agree not super important

**Comment by thiagohora** [comment]
Thank you, happy to revisit this in the future if needed

**Pull request merged**